nodebuilder/p2p/network.go

```diff
@@ -50,6 +64,28 @@ func (n Network) String() string {
 	return string(n)
 }
 
+// ChainIDFromNetwork parses the ChainID from the Network name.
+func ChainIDFromNetwork(net Network) ChainID {
+	if !strings.Contains(net.String(), "da") {
+		return ChainID(net.String())
+	}
+	// parse the ChainID from the custom Network name
+	networkName := strings.Split(net.String(), "-")
+	switch len(networkName) {
+	case 3:
+		// handles case where no suffix has been added to core consensus chainID
+		return ChainID(networkName[0])
+	case 4:
+		return ChainID(strings.Join(networkName[:2], "-"))
+	}
+	return ""
```

Comment:
This should never happen, but if it does, it will be caught as an error on Start inside of store.Init

Reply:
if this should never happen, can we simply panic here in order to show that smth went wrong?

Reply:
not applicable anymore